torch/testing/_internal/dist_utils.py

```diff
@@ -128,6 +148,24 @@ def wait_until_pending_users_flushed():
         num_pending_users = int(_rref_context_get_debug_info()["num_pending_users"])
     return
 
+def wait_until_pending_forks_flushed():
+    '''
+    The RRef protocol holds forks in a map and expects that all forks have been
+    deleted by the time shutdown is called. If not, this is considered an RRef
+    leak. When there are network failures and we retry RRef control messages,
+    the second try will happen some time after the first due to timeout +
+    backoff. Thus short-running tests may complete and call shutdown before the
+    retried RPCs attempt to delete forks. This function waits until the state
+    is cleaned up, and therefore is a more accurate test of whether retries
+    work in long-running programs with potential network failures.
+    '''
+    num_pending_users = int(_rref_context_get_debug_info()["num_forks"])
+    while num_pending_users != 0:
+        time.sleep(0.1)
+        num_pending_users = int(_rref_context_get_debug_info()["num_forks"])
+        print(num_pending_users)
```

Comment:
if you want to keep this print, add more information so someone seeing this on the terminal can understand what this number is.